### ISSUE TITLE
Avalonia 11.0 has bizarre reference assemblies, malformed Embedded PDB's which break source navigation in Visual Studio.

### ISSUE THREAD

**Issue by chuckries** [AvaloniaUI/Avalonia#12431]
**Describe the bug**
Following up in this repo on behalf of [this Visual Studio issue](https://developercommunity.visualstudio.com/t/Go-To-Definition-stops-working-for-Sourc/10428087).

The symptom is that F12 Go To Definition is not working for Source Link or embedded files, while it did in the 0.10.21 release. It appears that the 11.0 release started shipping reference assemblies in the NuGet packages. The reference assemblies claim to have embedded PDB's, but the embedded PDB is malformed. 

Here is a screenshot from ILSpy. The 11.0.2 net6.0 ref assembly for Avalonia.Controls claims to have an embedded pdb, but the embedded pdb contains no metadata (or the metadata is malformed).
![image](https://github.com/AvaloniaUI/Avalonia/assets/6598594/3aaf6fb1-13be-47d4-983c-6315ebbc5d9e)

The implementation assembly in the lib folder does not have an embedded PDB at all. 

I'm not sure what the intent is, but something seems off about these. And, as mentioned, the malformed embedded PDB is breaking source navigation in Visual Studio.

**To Reproduce**

1. In VS 2022, make a new C# Console App targeting net6.0 or above. 
2. Add a PackageReference to Avalonia 11.0.2
```
  <ItemGroup>
    <PackageReference Include="Avalonia" Version="11.0.2" />
  </ItemGroup>

```
3. Add a code reference to `Avalonia.Controls.Templates.IDataTemplate`
```cs
Avalonia.Controls.Templates.IDataTemplate foo = null;
```
4. Place caret on `IDataTemplate`
5. F12

**Expected behavior**
IDataTemplate.cs is opened either from SourceLink or a decopmiled file.

##Actual behavior** 
Nothing happens.

**If I change my package reference to version 0.10.21, and try the repro again, it obtains the file correctly from Source Link**

**Additional context**

**Comment by IanRawley** [comment]
I think this is a known issue and fixed in a recently merged PR. https://github.com/AvaloniaUI/Avalonia/pull/12410#issue-1831654386

**Comment by timunie** [comment]
@chuckries can you try using latest nightly? If the issue is solved there for you, please close this ticket.

**Comment by Mrxx99** [comment]
This seems to be fixed with 11.0.3 (works for me)

**Comment by timunie** [comment]
closing this due to positive feedback
